src/app/services/search.service.ts

```diff
@@ -1,17 +1,30 @@
 import { SearchResult } from './../models/searchResult';
+import { Artist } from '../models/artist';
+
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
-import { Observable } from 'rxjs';
+import { Observable, BehaviorSubject } from 'rxjs';
 
 @Injectable({
   providedIn: 'root'
 })
 export class SearchService {
-  baseURL = 'https://cors-anywhere.herokuapp.com/http://api.deezer.com/search?';
-  params = 'limit=2&';
+  private baseURL = 'https://cors-anywhere.herokuapp.com/http://api.deezer.com';
+  private params = '';
+  private dataSource = new BehaviorSubject<SearchResult>(new SearchResult());
```

Comment:
Actually you don't need any tmp storage here. Service should have a method to fetch something, maybe some operators in `.pipe` and that's it. It should fetch, transform and return value, not to store it. But these rules are in case of http services